PlayCover/Utils/URLHandler.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  URIHandler.swift
+//  PlayCover
+//
+//  Created by Venti on 14/02/2023.
+//
+
+import Foundation
+
+enum URLTypes: Int, Equatable {
+    case source
+    case keymap
+    case app
+}
+
+enum URLAction: Int, Equatable {
+    case add
+    case remove
+    case update
+    case install
+    case open
+}
+
+class URLObservable: ObservableObject {
+    @Published var url: String?
+    @Published var type: URLTypes?
+    @Published var action: URLAction?
+
+    public static var shared = URLObservable()
+}
+
+struct URLHandler {
+    public static var shared = URLHandler()
+
+    func processURL(url: URL) {
+        guard let urlComponenents = NSURLComponents(url: url, resolvingAgainstBaseURL: false),
+            let uriPath = urlComponenents.path,
+            let params = urlComponenents.queryItems else {
+                // Fall back to old url handler (for files)
+                if url.pathExtension == "ipa" {
+                    Installer.install(ipaUrl: url, export: false, returnCompletion: { _ in
+                    Task { @MainActor in
+                        AppsVM.shared.fetchApps()
+                        NotifyService.shared.notify(
+                            NSLocalizedString("notification.appInstalled", comment: ""),
+                            NSLocalizedString("notification.appInstalled.message", comment: "")
+                        )
+                    }})
+                }
+                return
+            }
+        // URI format: playcoverapp://<object>?action=<action>&<param>=<value>
+        // Example: playcoverapp://source?action=add&url=https://homebrew.playcover.io
+
+        // Switch case for main uri path
+        switch uriPath {
+        case "/source":
```

Comment:
Judging by the example above, I don't think the slash is intended in this case. Otherwise the url handler for the type source would have to be called with three slashes `playcover:///source?action=add&url=<source_url>`
```suggestion
        case "source":
```

Reply:
Actually, no. Keep in mind this is the format of a URL
`URI = scheme ":" ["//" authority] path ["?" query] ["#" fragment]`
If you only have `source`, the URL will actually be
`playcover:source?action=add&url=something`
`/source`, meanwhile, will actually make
`playcover://source?action=add&url=something`